warble/cli.cc

```diff
@@ -204,4 +206,49 @@ void CLI::Profile(std::string username) {
     std::cout << "Could not find profile for user " << username << std::endl;
     LOG(INFO) << "error: " << s.error_message();
   }
+}
+
+void CLI::Stream(std::string hashtag) {
+  ClientContext init_context;
+  StreamRequest s_req;
+  s_req.set_hashtag(hashtag);
+  EventRequest req;
+  int event_type = find_event_type_("stream");
+  req.set_event_type(event_type);
+  Any rx_payload, tx_payload;
+  tx_payload.PackFrom(s_req);
+  *(req.mutable_payload()) = tx_payload;
+  EventReply rep;
+  size_t num_warbles = 0;
+  Status s = func_->event(&init_context, req, &rep);
+  if ((s.ok())) {
+    rx_payload = rep.payload();
+    StreamReply s_rep;
+    rx_payload.UnpackTo(&s_rep);
+    num_warbles = s_rep.warbles_size();
+  } else {
+    num_warbles = 0;
+  }
+  while (1) {
```

Comment:
see earlier comment on polling

Reply:
I emailed the professor, and he said that we will not be graded on the efficiency of the streaming. Signalling changes to keys in the kvstore would definitely be better, but this works.